diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6cfb5c4..09b9c93 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,5 @@
 cmake_minimum_required(VERSION 2.6 FATAL_ERROR)
+if (${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}.${CMAKE_PATCH_VERSION} VERSION_GREATER 3.0.0)
+cmake_policy(SET CMP0042 NEW)
+endif()
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR}/cmake)
 set(CMAKE_BUILD_TYPE Debug)
 
diff --git a/src/functionals/pw92eps.h b/src/functionals/pw92eps.h
index 67c523e..10f946e 100644
--- a/src/functionals/pw92eps.h
+++ b/src/functionals/pw92eps.h
@@ -18,7 +18,7 @@ namespace pw92eps
   template<class num>
   static num omega(const densvars<num> &d)
   {
-    return (cbrt(2)/(cbrt(2)-1))*(d.a_43+d.b_43)*d.n_m13/d.n - 1/(cbrt(2)-1);
+    return (cbrt(2.0)/(cbrt(2.0)-1))*(d.a_43+d.b_43)*d.n_m13/d.n - 1/(cbrt(2.0)-1);
   }
 
   template<class num>
